scalyr_agent/tests/scalyr_client_test.py

```diff
@@ -34,14 +34,15 @@ def test_basic_case(self):
         request.set_client_time(1)
 
         self.assertEquals(request.total_events, 0)
-
         self.assertTrue(request.add_event(Event().set_message("eventOne"), timestamp=1L))
         self.assertTrue(request.add_event(Event().set_message("eventTwo"), timestamp=2L))
-
+        # get the next timestamp and assert for previous two
+        next_timestamp = request._AddEventsRequest__get_valid_timestamp(1L)
```

Comment:
I don't think we should be testing in this way.  You are asking for the timestamp after all of the `add_event` code was run.. and essentially just testing that the timestamps were assigned -1 and -2 of the next timestamp.  That's not really testing any interesting behavior.

I'd like to understand why you had to change this code at all.  Looking at Imron's relevant changes, it seems like this testing code should still work.  What values were you seeing?

Maybe we need some test method to reset the global timestamp counter or something.

Reply:
we were seeing nano seconds (i think) that were monotonically increasing and were absolute, not relative monotonically increasing values like 1L, 2L etc. The tests i tried to patch as mentioned before were hacky because i can't find a better way to have the method return relative ordering, and not absolute values. I'll let you or Imron decide that.

Reply:
but yeah, the tests were definitely broken.